  0%|          | 0/24.15 [00:00<?, ?it/s]  4%|▍         | 1/24.15 [00:00<00:07,  2.95it/s]  8%|▊         | 2/24.15 [00:00<00:05,  3.70it/s] 12%|█▏        | 3/24.15 [00:00<00:05,  4.03it/s] 17%|█▋        | 4/24.15 [00:01<00:04,  4.20it/s] 21%|██        | 5/24.15 [00:01<00:04,  4.30it/s] 25%|██▍       | 6/24.15 [00:01<00:04,  4.37it/s] 29%|██▉       | 7/24.15 [00:01<00:03,  4.41it/s] 33%|███▎      | 8/24.15 [00:01<00:03,  4.44it/s] 37%|███▋      | 9/24.15 [00:02<00:03,  4.46it/s] 41%|████▏     | 10/24.15 [00:02<00:03,  4.47it/s] 46%|████▌     | 11/24.15 [00:02<00:02,  4.42it/s] 50%|████▉     | 12/24.15 [00:02<00:02,  4.43it/s] 54%|█████▍    | 13/24.15 [00:03<00:02,  4.44it/s] 58%|█████▊    | 14/24.15 [00:03<00:02,  4.45it/s] 62%|██████▏   | 15/24.15 [00:03<00:02,  4.45it/s] 66%|██████▋   | 16/24.15 [00:03<00:01,  4.46it/s] 70%|███████   | 17/24.15 [00:03<00:01,  4.46it/s] 75%|███████▍  | 18/24.15 [00:04<00:01,  4.46it/s] 79%|███████▊  | 19/24.15 [00:04<00:01,  4.46it/s] 83%|████████▎ | 20/24.15 [00:04<00:00,  4.46it/s] 87%|████████▋ | 21/24.15 [00:04<00:00,  4.41it/s] 91%|█████████ | 22/24.15 [00:05<00:00,  4.40it/s] 95%|█████████▌| 23/24.15 [00:05<00:00,  4.41it/s] 99%|█████████▉| 24/24.15 [00:05<00:00,  4.42it/s]25it [00:05,  4.51it/s]                           
  0%|          | 0/9 [00:00<?, ?it/s] 56%|█████▌    | 5/9 [00:00<00:00, 41.46it/s]100%|██████████| 9/9 [00:00<00:00, 36.69it/s]
/project2/ruishanl_1185/Tumor_Segmentation_Summer2025_XWDR/Xiwen/radiogenomics/utils/preprocess_original.py:405: RuntimeWarning: invalid value encountered in divide
  real_resize_factor = new_shape / np.array(image_np.shape)
Traceback (most recent call last):
  File "/project2/ruishanl_1185/Tumor_Segmentation_Summer2025_XWDR/Xiwen/radiogenomics/utils/preprocess_original.py", line 421, in normalize
    image = tio.Resize(out_shape, image_interpolation=interp)(image)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home1/xiwenc/.local/lib/python3.11/site-packages/torchio/transforms/transform.py", line 163, in __call__
    transformed = self.apply_transform(subject)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home1/xiwenc/.local/lib/python3.11/site-packages/torchio/transforms/preprocessing/spatial/resize.py", line 65, in apply_transform
    resampled = resample(subject)
                ^^^^^^^^^^^^^^^^^
  File "/home1/xiwenc/.local/lib/python3.11/site-packages/torchio/transforms/transform.py", line 163, in __call__
    transformed = self.apply_transform(subject)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home1/xiwenc/.local/lib/python3.11/site-packages/torchio/transforms/preprocessing/spatial/resample.py", line 193, in apply_transform
    self._set_resampler_reference(
  File "/home1/xiwenc/.local/lib/python3.11/site-packages/torchio/transforms/preprocessing/spatial/resample.py", line 267, in _set_resampler_reference
    self._set_resampler_from_spacing(resampler, target, floating_sitk)
  File "/home1/xiwenc/.local/lib/python3.11/site-packages/torchio/transforms/preprocessing/spatial/resample.py", line 279, in _set_resampler_from_spacing
    target_spacing = self._parse_spacing(target)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home1/xiwenc/.local/lib/python3.11/site-packages/torchio/transforms/preprocessing/spatial/resample.py", line 123, in _parse_spacing
    raise ValueError(message)
ValueError: Spacing must be strictly positive, not "[0.         0.99609375 1.27148438]"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/project2/ruishanl_1185/Tumor_Segmentation_Summer2025_XWDR/Xiwen/radiogenomics/test.py", line 34, in <module>
    pre.radiogenomics()
  File "/project2/ruishanl_1185/Tumor_Segmentation_Summer2025_XWDR/Xiwen/radiogenomics/utils/preprocess_original.py", line 279, in radiogenomics
    self.extract_lungs(patient, patient_ct, seg_np)
  File "/project2/ruishanl_1185/Tumor_Segmentation_Summer2025_XWDR/Xiwen/radiogenomics/utils/preprocess_original.py", line 311, in extract_lungs
    lungs_seg = self.normalize(lungs_seg, is_label=True)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project2/ruishanl_1185/Tumor_Segmentation_Summer2025_XWDR/Xiwen/radiogenomics/utils/preprocess_original.py", line 439, in normalize
    tensor = resample_numpy(tensor, spacing, space)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project2/ruishanl_1185/Tumor_Segmentation_Summer2025_XWDR/Xiwen/radiogenomics/utils/preprocess_original.py", line 406, in resample_numpy
    image_resampled = ndimage.zoom(image_np, real_resize_factor, order=0 if is_label else 3)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home1/xiwenc/.local/lib/python3.11/site-packages/scipy/ndimage/_interpolation.py", line 821, in zoom
    [int(round(ii * jj)) for ii, jj in zip(input.shape, zoom)])
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home1/xiwenc/.local/lib/python3.11/site-packages/scipy/ndimage/_interpolation.py", line 821, in <listcomp>
    [int(round(ii * jj)) for ii, jj in zip(input.shape, zoom)])
         ^^^^^^^^^^^^^^
ValueError: cannot convert float NaN to integer
